massa-grpc/src/config.rs

```diff
@@ -23,6 +23,10 @@ pub struct GrpcConfig {
     pub enable_tls: bool,
     /// whether to enable mTLS (requires `enable_tls` to be true)
     pub enable_mtls: bool,
+    /// whether to generate a self-signed certificate if none is provided(ignored if `enable_tls` is false)
+    pub generate_self_signed_certificates: bool,
+    /// Subject Alternative Names is an extension in X.509 certificates that allows a certificate to specify additional subject identifiers. It is used to support alternative names for a subject, other than its primary Common Name (CN), which is typically used to represent the primary domain name.
+    pub subject_alt_names: Vec<String>,
```

Comment:
I assume this will remain empty for most nodes? Is it needed if you want to access it with both the IP:port and a domain name?
Maybe add an example use case here, if it's not too long to explain!